elfi/methods/bo/acquisition.py

```diff
@@ -547,6 +549,172 @@ def _evaluate_logpdf(theta):
+        sigma_proposals_metropolis : array_like, optional
+            Standard deviation proposals for tuning the metropolis sampler.
+
+        """
+        super(ExpIntVar, self).__init__(quantile_eps, *args, **opts)
+        self.name = 'exp_int_var'
+        self.label_fn = 'Expected Loss'
+        self._n_imp_samples = n_imp_samples
+        self._iter_imp = iter_imp
+
+        # Initialising the RandMaxVar density for the importance sampling.
+        self.density_is = RandMaxVar(model=self.model,
+                                     prior=self.prior,
+                                     n_inits=self.n_inits,
+                                     seed=self.seed,
+                                     quantile_eps=self.quantile_eps,
+                                     sampler=sampler,
+                                     n_samples=n_samples,
+                                     sigma_proposals_metropolis=sigma_proposals_metropolis)
+
+    def acquire(self, n, t):
+        """Acquire a batch of acquisition points.
+
+        Parameters
+        ----------
+        n : int
+            Number of acquisitions.
+        t : int
+            Current iteration.
+
+        Returns
+        -------
+        array_like
+            Coordinates of the yielded acquisition points.
+
+        """
+        logger.debug('Acquiring the next batch of %d values', n)
+        gp = self.model
+        self.sigma_n = gp.noise
+
+        # Updating the discrepancy threshold.
+        self.eps = np.percentile(gp.Y, self.quantile_eps * 100)
+
+        # Performing the importance sampling step every self._iter_imp iterations.
+        if t % self._iter_imp == 0:
+            self.samples_imp = self.density_is.acquire(self._n_imp_samples)
+            self.mean_imp, self.var_imp = gp.predict(self.samples_imp, noiseless=True)
+            self.phi_imp = ss.norm.cdf(self.eps, loc=self.mean_imp.T,
+                                       scale=np.sqrt(self.sigma_n + self.var_imp.T))
+
+        # Pre-calculating the omegas_imp and priors_imp terms to be used in the evaluate function.
+        omegas_imp_unnormalised = (1 / self.density_is.evaluate(self.samples_imp)).T
+        self.priors_imp = (self.prior.pdf(self.samples_imp)**2)[np.newaxis, :]
+        self.omegas_imp = omegas_imp_unnormalised / \
+            np.sum(omegas_imp_unnormalised, axis=1)[:, np.newaxis]
+
+        # Initialising the attributes used in the evaluate function.
+        self.thetas_old = np.array(gp.X)
+        self._K = gp._gp.kern.K
+        self.K = self._K(self.thetas_old, self.thetas_old) + \
+            self.sigma_n * np.identity(self.thetas_old.shape[0])
+        self.k_imp_old = self._K(self.samples_imp, self.thetas_old).T
+
+        # Obtaining the location where the expected loss is minimised.
+        # Note: The gradient is computed numerically as GPy currently does not
+        # provide the derivative computations used in Järvenpää et al., 2017.
```

Comment:
maybe add "directly" in front of provide, we could probably implement it ourselves but that might require quite a lot of work so using finite differences seems a reasonable approach for now.

Reply:
Added.